cpp/harvest/marc_issn_lookup.cc

```diff
@@ -0,0 +1,190 @@
+                MARC::Subfields subfields(field.getSubfields());
+                std::string subfield_a(subfields.getFirstSubfieldWithCode('a'));
+                std::string subfield_b(subfields.getFirstSubfieldWithCode('b'));
+                if (subfield_a != "" && subfield_b != "")
+                    t_ = subfield_a + " " + subfield_b;
+                else if (subfield_a != "" && subfield_b == "")
+                    t_ = subfield_a;
+                else if (subfield_a == "" && subfield_b != "")
+                    t_ = subfield_b;
+                else
+                    t_ = "";
+            }
+
+            if (field.getTag() == "300")
+                ((field.getFirstSubfieldWithCode('a') == "Online-Ressource") ? is_online_ = true : is_online_ = false);
+
+
+            if (field.getTag() == "773")
+                x_ = field.getFirstSubfieldWithCode('x');
+        }
+
+        i_ = "In:";
+    }
+};
+
+
+void UpdateSubfield(MARC::Subfields &subfields, const SubFieldInfo &sub_field_info) {
+    if (!subfields.replaceFirstSubfield('i', sub_field_info.i_))
+        subfields.addSubfield('i', sub_field_info.i_);
+    if (!subfields.replaceFirstSubfield('x', sub_field_info.x_))
+        subfields.addSubfield('x', sub_field_info.x_);
+    if (!subfields.replaceFirstSubfield('w', sub_field_info.w_))
+        subfields.addSubfield('w', sub_field_info.w_);
+
+    if (sub_field_info.t_ != "") {
+        if (!subfields.replaceFirstSubfield('t', sub_field_info.t_))
+            subfields.addSubfield('t', sub_field_info.t_);
+    }
+}
+
+void UpdateJournalValidity(std::map<std::string, SubFieldInfo> &journal_cache) {
+    // check whether the issn is occur more than once
+    for (auto sfi : journal_cache) {
+        if (sfi.second.online_version_counter_ == 1)
+            journal_cache[sfi.first].is_valid_ = true;
+        else if (sfi.second.online_version_counter_ > 1)
+            journal_cache[sfi.first].is_valid_ = false;
+        else if (sfi.second.online_version_counter_ == 0)
+            if (sfi.second.printed_version_counter_ == 1)
+                journal_cache[sfi.first].is_valid_ = true;
+            else
+                journal_cache[sfi.first].is_valid_ = false;
+    }
+}
+
+std::map<std::string, SubFieldInfo> JournalCacheBuilder(const std::string &input_journal_filename) {
+    std::map<std::string, SubFieldInfo> journal_cache;
+    auto input_journal_file(MARC::Reader::Factory(input_journal_filename));
+    int record_counter(0);
+
+    std::cout << "Build a cache for journal \n";
+    while (MARC::Record record = input_journal_file->read()) {
+        SubFieldInfo sub_field_info_of_record;
+        sub_field_info_of_record.Builder(record);
+
+        // if sub_info is exist in the cache
+        if (auto sub_info_search = journal_cache.find(sub_field_info_of_record.x_); sub_info_search != journal_cache.end()) {
+            if (journal_cache[sub_field_info_of_record.x_].is_online_) {
```

Comment:
Maybe we should move the increasing logic of the statistics out of these blocks to avoid similar code in separate branches?